EvolvSwiftSDK/EvolvSwiftSDK/Sources/Protocols/EvolvClient.swift

```diff
@@ -41,66 +33,107 @@ public protocol EvolvClient {
     ///  Method produces a contamination event which will contaminate the
     /// participant's allocation. Method will not do anything in the event
     /// that the allocation timed out or failed.
-    /// - Parameter details: Optional. Information on the reason for contamination. If provided, the object should contain a reason. Optionally, a 'details' value should be included for extra debugging info
-    /// - Parameter allExperiments: If true, the user will be excluded from all optimizations, including optimization not applicable to this page
-    func contaminate(for details: AnyObject?, _ allExperiments: Bool)
-    
-    /// Destroy the client and its dependencies.
-    func destroy()
+    func contaminate()
     
-    /// Emits a generic event to be recorded by Evolv.
-    ///
-    /// Sends an event to Evolv to be recorded and reported upon.
-    ///
-    /// - Parameter key: The identifier of the event.
-    func emit(for key: String)
     
-    /// Force all beacons to transmit.
-    func flush()
     
     /// Get the value of a specified key.
     /// - Parameter forKey: The identifier of the event.
     func get(value forKey: String)
     
-    /// Check all active keys that start with the specified prefix.
-    /// - Parameter prefix: The prefix of the keys to check.
-    func getActiveKeys(for prefix: String)
-    
-    /// Get the configuration for a specified key.
-    /// - Parameter key: The key to retrieve the configuration for.
-    func getConfig(for key: String)
-    
     /// Initializes the client with required context information.
     /// - Parameter uid: A globally unique identifier for the current participant.
     /// - Parameter remoteContext: A map of data used for evaluating context predicates and analytics.
     /// - Parameter localContext: A map of data used only for evaluating context predicates.
     func initialize(uid: String, remoteContext: [String: AnyObject], localContext: [String: AnyObject])
     
+    
+    
+    /// Reevaluates the current context.
+    func reevaluateContext()
+}
+
+// MARK: - Default implementation and optional protocol methods
+extension EvolvClient {
+    
+    /// Check all active keys that start with the specified prefix.
+    /// - Parameter prefix: The prefix of the keys to check.
+    func getActiveKeys(for prefix: String) -> Bool {
```

Comment:
It is an optional implementation for protocols so that it couldn't effect other classes conforming to this protocol and will be changed in the future pull requests.